src/Microsoft.AspNet.Mvc.Razor/Diagnostics/IRuntimeCompilationException.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Framework.Runtime;
+
+namespace Microsoft.AspNet.Diagnostics
+{
+    /// <summary>
+    /// Specifies the contracts for an exception that represents a compilation failure at runtime.
+    /// </summary>
+    [AssemblyNeutral]
+    public interface IRuntimeCompilationException
```

Comment:
Using this for application compilation errors might mean this needs to become more general - for instance the assumption here is that we're compiling a single file.